tests/Collection.test.ts

```diff
@@ -44,33 +45,94 @@ describe("Collection", () => {
   });
 
   describe("loadCollection", () => {
-    it("should parse dataset paths correctly", async () => {
-      const url = "https://e.com/collection.json/";
-      const collectionJson = [
-        { name: "dataset1", path: "dir1" },
-        { name: "dataset2", path: "dir2/nested/some_dir/" },
-        { name: "dataset3", path: "dir3/data.json" },
-        { name: "dataset4", path: "https://b.com/dataset4/" },
-        { name: "dataset5", path: "https://b.com/dataset5/custom.json" },
-      ];
-
-      const mockFetch = makeMockFetchMethod("https://e.com/collection.json", collectionJson);
-      const collection = await Collection.loadCollection(url, mockFetch);
-
-      expect(collection.getDatasetKeys().length).to.equal(5);
-      expect(collection.getAbsoluteDatasetPath("dataset1")).to.equal("https://e.com/dir1/manifest.json");
-      expect(collection.getAbsoluteDatasetPath("dataset2")).to.equal(
-        "https://e.com/dir2/nested/some_dir/manifest.json"
-      );
-      expect(collection.getAbsoluteDatasetPath("dataset3")).to.equal("https://e.com/dir3/data.json");
-      expect(collection.getAbsoluteDatasetPath("dataset4")).to.equal("https://b.com/dataset4/manifest.json");
-      expect(collection.getAbsoluteDatasetPath("dataset5")).to.equal("https://b.com/dataset5/custom.json");
-    });
+    const datasets = [
+      { name: "dataset1", path: "dir1" },
+      { name: "dataset2", path: "dir2/nested/some_dir/" },
+      { name: "dataset3", path: "dir3/data.json" },
+      { name: "dataset4", path: "https://b.com/dataset4/" },
+      { name: "dataset5", path: "https://b.com/dataset5/custom.json" },
+    ];
+
+    const collectionFormats: { version: string; collectionJson: object }[] = [
+      { version: "v0.0.0", collectionJson: datasets },
+      {
+        version: "v1.1.0",
+        collectionJson: {
+          datasets: datasets,
+          metadata: {
+            name: "c_name",
+            description: "c_description",
+            author: "c_author",
+            collectionVersion: "c_collectionVersion",
+            lastModified: "2024-01-01T00:00:00Z",
+            dateCreated: "2023-01-01T00:00:00Z",
+            revision: 15,
+            writerVersion: "v1.1.0",
+          },
+        },
+      },
+    ];
+
+    for (const { version, collectionJson } of collectionFormats) {
+      describe(version, () => {
+        it("collects datasets", async () => {
+          const url = "https://e.com/collection.json";
+          const mockFetch = makeMockFetchMethod(url, collectionJson);
+          const collection = await Collection.loadCollection(url, mockFetch);
+
+          expect(collection.getDatasetKeys().length).to.equal(5);
```

Comment:
small thing: but couldn't you do:
```suggestion
          expect(collection.getDatasetKeys().length).to.equal(datasets.length);
```
?
just in case you add more data to your mock data